drud/ddev-ui
# Issue 32 initial testing framework

## The Problem/Issue/Bug:
There was no testing framework in place for DDEV-UI

## How this PR Solves The Problem:
Added Mocha testing framework and the modules needed to mock the response from ddevCLI

## Manual Testing Instructions:
`npm install` to ensure all dependencies are installed, then `npm test` to run the automated testing. The current test simply verifies that the parsing of the ddev list cli output is properly converted to a json object that the rest of the application uses. Unit tests to address individual functionality will be built on this base and committed with the feature branches for that particular feature as part of acceptance criteria.

To break the test intentionally, change data in either the sitesList variable on line 5 or in the expectedSitesList variable on line 16 and rerun. 

## Automated Testing Overview:
1 test to demonstrate a working test platform:
ddevShell.list()

## Related Issue Link(s):
https://github.com/drud/ddev-ui/issues/32
